### PR TITLE
Convert lib server methods to JS

### PR DESCRIPTION
**Pull request by Kiran-Rao**
@RocketChat/core 

Closes #ISSUE_NUMBER

### CODE REVIEW

File: packages/rocketchat-lib/server/methods/setRealName.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+Meteor.methods({
+	setRealName(name) {
+
+		check(name, String);
+
+		if (!Meteor.userId()) {
+			throw new Meteor.Error('error-invalid-user', 'Invalid user', { method: 'setRealName' });
+		}
+
+		let user = Meteor.user();
```
Comment (rodrigok) [inline packages/rocketchat-lib/server/methods/setRealName.js]: Can you change here to `const`?

File: packages/rocketchat-lib/server/methods/setUsername.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+		const { joinDefaultChannelsSilenced } = param;
+		check(username, String);
+
+		if (!Meteor.userId()) {
+			throw new Meteor.Error('error-invalid-user', 'Invalid user', { method: 'setUsername' });
+		}
+
+		const user = Meteor.user();
+
+		if ((user.username != null) && !RocketChat.settings.get('Accounts_AllowUsernameChange')) {
```
Comment (rodrigok) [inline packages/rocketchat-lib/server/methods/setUsername.js]: Can you change
```
if ((user.username != null) && !RocketChat.settings.get('Accounts_AllowUsernameChange')) {
```
by
```
if (user.username && !RocketChat.settings.get('Accounts_AllowUsernameChange')) {
```

File: packages/rocketchat-lib/server/methods/setUsername.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+			}
+		} else if (!RocketChat.checkUsernameAvailability(username)) {
+			throw new Meteor.Error('error-field-unavailable', `<strong>${_.escape(username)}</strong> is already in use :(`, { method: 'setUsername', field: username });
+		}
+
+		if (!RocketChat.setUsername(user._id, username)) {
+			throw new Meteor.Error('error-could-not-change-username', 'Could not change username', { method: 'setUsername' });
+		}
+
+		if (user.username == null) {
```
Comment (rodrigok) [inline packages/rocketchat-lib/server/methods/setUsername.js]: Can you change `if (user.username == null) {` by `if (!user.username) {`?

File: packages/rocketchat-lib/server/methods/updateMessage.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+		check(message, Match.ObjectIncluding({_id:String}));
+
+		if (!Meteor.userId()) {
+			throw new Meteor.Error('error-invalid-user', 'Invalid user', { method: 'updateMessage' });
+		}
+
+		const originalMessage = RocketChat.models.Messages.findOneById(message._id);
+
+		if (!originalMessage && !originalMessage._id) {
```
Comment (rodrigok) [inline packages/rocketchat-lib/server/methods/updateMessage.js]: This line will fail if originalMessage is undefined
Can you change
`if (!originalMessage && !originalMessage._id) {`
by
`if (!originalMessage || !originalMessage._id) {`

File: packages/rocketchat-lib/server/methods/updateMessage.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+			return;
+		}
+
+		const hasPermission = RocketChat.authz.hasPermission(Meteor.userId(), 'edit-message', message.rid);
+		const editAllowed = RocketChat.settings.get('Message_AllowEditing');
+		const editOwn = originalMessage && originalMessage.u && (originalMessage.u._id === Meteor.userId());
```
Comment (rodrigok) [inline packages/rocketchat-lib/server/methods/updateMessage.js]: If originalMessage does not exist the code will return on line 13, so you don't need to check for it again.

You can use only
`const editOwn = originalMessage.u && originalMessage.u._id === Meteor.userId();`

File: packages/rocketchat-lib/server/methods/updateMessage.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+		if (!hasPermission && (!editAllowed || !editOwn)) {
+			throw new Meteor.Error('error-action-not-allowed', 'Message editing not allowed', { method: 'updateMessage', action: 'Message_editing' });
+		}
+
+		const blockEditInMinutes = RocketChat.settings.get('Message_AllowEditing_BlockEditInMinutes');
+		if ((blockEditInMinutes != null) && (blockEditInMinutes !== 0)) {
```
Comment (rodrigok) [inline packages/rocketchat-lib/server/methods/updateMessage.js]: Can you replace by `if (Match.test(blockEditInMinutes, Number) && blockEditInMinutes !== 0) {`?

File: packages/rocketchat-lib/server/methods/updateMessage.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+			let currentTsDiff, msgTs;
+			if (originalMessage.ts != null) { msgTs = moment(originalMessage.ts); }
```
Comment (rodrigok) [inline packages/rocketchat-lib/server/methods/updateMessage.js]: `if (Match.test(originalMessage.ts, Number))`

File: packages/rocketchat-lib/server/methods/updateMessage.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+			if (msgTs != null) { currentTsDiff = moment().diff(msgTs, 'minutes'); }
```
Comment (rodrigok) [inline packages/rocketchat-lib/server/methods/updateMessage.js]: Replace `if (msgTs != null)` by `if (msgTs)`

File: packages/rocketchat-lib/server/methods/updateMessage.js [hunk 9]
```diff
@@ -0,0 +1,36 @@
+Meteor.methods({
```
Comment (rodrigok) [inline packages/rocketchat-lib/server/methods/updateMessage.js]: Why did you remove the moment import? We can remove it from `.eslint` globals if you need

Comment (Kiran-Rao) [inline packages/rocketchat-lib/server/methods/updateMessage.js]: My mistake.
